calibrate.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python
+
+import matplotlib.pyplot as plt
+import numpy as np
+import time
+import cv2
+from real.camera import Camera
+from robot import Robot
+from scipy import optimize  
+from mpl_toolkits.mplot3d import Axes3D  
+
+
+# User options (change me)
+# --------------- Setup options ---------------
+tcp_host_ip = '100.127.7.223' # IP and port to robot arm as TCP client (UR5)
+tcp_port = 30002
+rtc_host_ip = '100.127.7.223' # IP and port to robot arm as real-time client (UR5)
+rtc_port = 30003
+workspace_limits = np.asarray([[0.3, 0.748], [0.05, 0.4], [-0.2, -0.1]]) # Cols: min max, Rows: x y z (define workspace limits in robot coordinates)
+calib_grid_step = 0.05
+checkerboard_offset_from_tool = [0,-0.13,0.02]
+tool_orientation = [-np.pi/2,0,0] # [0,-2.22,2.22] # [2.22,2.22,0]
+# ---------------------------------------------
+
+
+# Construct 3D calibration grid across workspace
+gridspace_x = np.linspace(workspace_limits[0][0], workspace_limits[0][1], 1 + (workspace_limits[0][1] - workspace_limits[0][0])/calib_grid_step)
+gridspace_y = np.linspace(workspace_limits[1][0], workspace_limits[1][1], 1 + (workspace_limits[1][1] - workspace_limits[1][0])/calib_grid_step)
+gridspace_z = np.linspace(workspace_limits[2][0], workspace_limits[2][1], 1 + (workspace_limits[2][1] - workspace_limits[2][0])/calib_grid_step)
+calib_grid_x, calib_grid_y, calib_grid_z = np.meshgrid(gridspace_x, gridspace_y, gridspace_z)
```

Comment:
I believe this is the part where the paper mentions about calibration. 
_"the location of a checkerboard pattern taped onto the gripper. The calibration optimizes for extrinsics as the robot moves the gripper over a grid of 3D locations (predefined with respect to robot coordinates) within the camera field of view"_